lib/jsonapi_compliable/deserializer.rb

```diff
@@ -46,16 +46,22 @@
 #   { type: 'authors', method: :create, temp_id: 'abc123' }
 class JsonapiCompliable::Deserializer
   # @param payload [Hash] The incoming payload with symbolized keys
+  # @param method [String] The method that the request was sent with
   # @param env [Hash] the Rack env (e.g. +request.env+).
-  def initialize(payload, env)
+  def initialize(payload, method, env)
     @payload = payload
     @payload = @payload[:_jsonapi] if @payload.has_key?(:_jsonapi)
+    @method = method
     @env = env
   end
 
   # @return [Hash] the raw :data value of the payload
   def data
```

Comment:
All looks good, except I was hoping we could do something like `validate_payload!` within the constructor, to keep the logic out of the accessor. The intent is to eventually do more complex validations, possibly using jsonapi-rb's parser.